agent/exec/containerd/executor.go

```diff
@@ -0,0 +1,107 @@
+package containerd
```

Comment:
With this we now have `github.com/docker/swarmkit/agent/exec/container` and `github.com/docker/swarmkit/agent/exec/containerd` which differ in a single character suffix. Perhaps we should consider renaming the former to `github.com/docker/swarmkit/agent/exec/dockerengine` (or `.../dockerapi` or whatever suites)?

WRT my comment in the PR text "perhaps be considering a new spec type specifically for containerd" maybe it is also worth considering renaming the `ContainerSpec` gRPC type too?